File: playground/javascript/node/ingestion.ts
import { ingestionClient } from '@algolia/ingestion';
import { ApiError } from '@algolia/client-common';
import dotenv from 'dotenv';

dotenv.config({ path: '../../.env' });

const appId = process.env.ALGOLIA_APPLICATION_ID || '**** APP_ID *****';
const apiKey =
  process.env.ALGOLIA_ADMIN_KEY ||
  '**** ADMIN_KEY *****';

// Init client with appId and apiKey
const client = ingestionClient(appId, apiKey, 'us', {authMode: 'WithinHeaders', hosts: [{url: 'staging-data.us.algolia.com', accept: 'readWrite', protocol: 'https'}]});

async function testIngestion() {
  try {
    const res = await client.getAuthentications();

    console.log(`[OK]`, res);
  } catch (e) {
    if (e instanceof ApiError) {
      return console.log(`[${e.status}] ${e.message}`, e.stackTrace, e);
    }

    console.log('[ERROR]', e);
  }
}

testIngestion();

File: playground/go/ingestion.go
package main

import (
	"fmt"

	"github.com/algolia/algoliasearch-client-go/v4/algolia/ingestion"
)

func testIngestion(appID, apiKey string) int {
	ingestionClient := ingestion.NewClient(appID, apiKey, ingestion.US)

	listAuthenticationsResponse, err := ingestionClient.GetAuthentications(
		ingestionClient.NewApiGetAuthenticationsRequest().WithItemsPerPage(2),
	)
	if err != nil {
		fmt.Printf("request error with GetAuthentications: %v\n", err)
		return 1
	}

	printResponse(listAuthenticationsResponse)

	return 0
}
